src/coreComponents/dataRepository/BufferOps_inline.hpp

```diff
@@ -1519,16 +1530,68 @@ Unpack( buffer_unit_type const * & buffer,
       }
     }
 
-    // insert known local indices into the set of indices
-    //  related to the local index
-    localIndex const numUniqueMapped = LvArray::sortedArrayManipulation::makeSortedUnique( mapped.begin(), mapped.end() );
-    var.insertIntoSet( li, mapped.begin(), mapped.begin() + numUniqueMapped );
+    // insert known local indices into the set of indices related to the local index
+    mapped.resize( LvArray::sortedArrayManipulation::makeSortedUnique( mapped.begin(), mapped.end() ) );
+    std::set_difference( mapped.begin(), mapped.end(), var[li].begin(), var[li].end(), std::back_inserter( mappedNew ) );
+    localIndex const numNewValues = LvArray::integerConversion< localIndex >( mappedNew.size() );
+
+    // check if we have enough capacity to insert new indices
+    if( numNewValues <= var.capacityOfSet( li ) - var.sizeOfSet( li ) )
+    {
+      // all good, just insert new entries
+      var.insertIntoSet( li, mappedNew.begin(), mappedNew.end() );
+    }
+    else
+    {
+      // need to stash away and rebuild the map later
+      localIndex const k = indiciesToInsert.size();
+      indiciesToInsert.emplace_back( li );
+      valuesToInsert.appendSet( numNewValues );
+      valuesToInsert.insertIntoSet( k, mappedNew.begin(), mappedNew.end() );
+    }
 
-    // insert unknown global indices related to the local index
-    //  into an additional mapping to resolve externally
-    localIndex const numUniqueUnmapped = LvArray::sortedArrayManipulation::makeSortedUnique( unmapped.begin(), unmapped.end() );
-    unmappedGlobalIndices[li].insert( unmapped.begin(), unmapped.begin() + numUniqueUnmapped );
+    // insert unknown global indices related to the local index into an additional mapping to resolve externally
+    unmapped.resize( LvArray::sortedArrayManipulation::makeSortedUnique( unmapped.begin(), unmapped.end() ) );
+    unmappedGlobalIndices[li].insert( unmapped.begin(), unmapped.end() );
```

Comment:
I'm just resizing `unmapped` to the range of unique values after duplicates have been moved to the end. Old code did not shrink but used number of unique values to specify a subrange to be inserted. Both are equivalent and identical in terms of performance, I just prefer this way stylistically since it follows the [remove-erase idiom](https://en.wikipedia.org/wiki/Erase%E2%80%93remove_idiom).